torch/_inductor/fx_passes/split_cat.py

```diff
@@ -78,5 +79,7 @@ def _split_cat_init():
         CallMethod("split", Arg(), Arg(), dim=Arg(), _users=MULTIPLE),
         CallMethod("split", Arg(), split_size=Arg(), dim=Arg(), _users=MULTIPLE),
     ]:
-        pattern = NormalizeSplit(pattern=pattern, extra_check=lambda arg: True)
+        pattern = NormalizeSplit(
+            pattern=pattern, extra_check=config_flag("split_cat_fx_passes")
```

Comment:
My only critique for this is loss of efficiency. We will now unnecessarily match a pattern, which could've been avoided if we didn't even register the pattern (if the flag was false)

Reply:
The old behavior was actually slightly wrong, since config flags can change dynamically after `lazy_init()` gets called.

If you:
1) Compiled something and triggered lazy_init()
2) Changed the value of `split_cat_fx_passes`

Then the prior value of `split_cat_fx_passes` would still get used and the new one ignored.